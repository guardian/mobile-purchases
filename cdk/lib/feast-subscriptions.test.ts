import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import {FeastSubscriptions} from "./feast-subscriptions";
//import { CloudwatchLogsManagement } fro;

describe('The FeastSubscriptions stack', () => {
    it('matches the snapshot', () => {
        const app = new App();
        const codeStack = new FeastSubscriptions(app, 'feast-subscriptions-CODE',{ stack: 'mobile', stage: 'CODE' });
        const prodStack = new FeastSubscriptions(app, 'feast-subscriptions-PROD',{ stack: 'mobile', stage: 'PROD' });

        expect(Template.fromStack(codeStack).toJSON()).toMatchSnapshot();
        expect(Template.fromStack(prodStack).toJSON()).toMatchSnapshot();
    });
});