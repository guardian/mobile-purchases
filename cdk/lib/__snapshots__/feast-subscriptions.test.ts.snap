// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FeastSubscriptions stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuSnsLambdaExperimental",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "feastsubscriptionsCODEdatabaselambdaTopicNameA7BAF410": {
      "Value": {
        "Fn::GetAtt": [
          "feastsubscriptionsCODEsnsqueueF2BD20CE",
          "QueueName",
        ],
      },
    },
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "feastsubscriptionsCODEdatabaselambda191E77D1": {
      "DependsOn": [
        "feastsubscriptionsCODEdatabaselambdaServiceRoleDefaultPolicy529EF1CD",
        "feastsubscriptionsCODEdatabaselambdaServiceRoleB1801835",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "mobile/CODE/feast-subscriptions-CODE-database-lambda/feast-subscriptions.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "feast-subscriptions-CODE-database-lambda",
            "STACK": "mobile",
            "STAGE": "CODE",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "feastsubscriptionsCODEdatabaselambdaServiceRoleB1801835",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-subscriptions-CODE-database-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "feastsubscriptionsCODEdatabaselambdaAllowInvokefeastsubscriptionsCODEfeastsubscriptionsCODEdatabaselambdaSnsExistingIncomingEventsTopic094866AB74B52CC0": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "feastsubscriptionsCODEdatabaselambda191E77D1",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastsubscriptionsCODEsnsqueueF2BD20CE",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "feastsubscriptionsCODEdatabaselambdaServiceRoleB1801835": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-subscriptions-CODE-database-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "feastsubscriptionsCODEdatabaselambdaServiceRoleDefaultPolicy529EF1CD": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/mobile/CODE/feast-subscriptions-CODE-database-lambda/feast-subscriptions.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/mobile/feast-subscriptions-CODE-database-lambda",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/CODE/mobile/feast-subscriptions-CODE-database-lambda/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "feastsubscriptionsCODEdatabaselambdaServiceRoleDefaultPolicy529EF1CD",
        "Roles": [
          {
            "Ref": "feastsubscriptionsCODEdatabaselambdaServiceRoleB1801835",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "feastsubscriptionsCODEdatabaselambdafeastsubscriptionsCODEdatabaselambdaSnsExistingIncomingEventsTopic2F7C96D8": {
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "feastsubscriptionsCODEdatabaselambda191E77D1",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "Region": {
          "Ref": "AWS::Region",
        },
        "TopicArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastsubscriptionsCODEsnsqueueF2BD20CE",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "feastsubscriptionsCODEsnsqueueF2BD20CE": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "feast-subscriptions-CODE-queue",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "CODE",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;

exports[`The FeastSubscriptions stack matches the snapshot 2`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuDistributionBucketParameter",
      "GuSnsLambdaExperimental",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "feastsubscriptionsPRODdatabaselambdaTopicName3ED11F02": {
      "Value": {
        "Fn::GetAtt": [
          "feastsubscriptionsPRODsnsqueue58434470",
          "QueueName",
        ],
      },
    },
  },
  "Parameters": {
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "feastsubscriptionsPRODdatabaselambda7E087BB6": {
      "DependsOn": [
        "feastsubscriptionsPRODdatabaselambdaServiceRoleDefaultPolicyCAE476A6",
        "feastsubscriptionsPRODdatabaselambdaServiceRoleD4C1286B",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "mobile/PROD/feast-subscriptions-PROD-database-lambda/feast-subscriptions.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "feast-subscriptions-PROD-database-lambda",
            "STACK": "mobile",
            "STAGE": "PROD",
          },
        },
        "Handler": "index.handler",
        "MemorySize": 512,
        "Role": {
          "Fn::GetAtt": [
            "feastsubscriptionsPRODdatabaselambdaServiceRoleD4C1286B",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-subscriptions-PROD-database-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "feastsubscriptionsPRODdatabaselambdaAllowInvokefeastsubscriptionsPRODfeastsubscriptionsPRODdatabaselambdaSnsExistingIncomingEventsTopicBCDBB18B356B5165": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "feastsubscriptionsPRODdatabaselambda7E087BB6",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastsubscriptionsPRODsnsqueue58434470",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "feastsubscriptionsPRODdatabaselambdaServiceRoleD4C1286B": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "feast-subscriptions-PROD-database-lambda",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "feastsubscriptionsPRODdatabaselambdaServiceRoleDefaultPolicyCAE476A6": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/mobile/PROD/feast-subscriptions-PROD-database-lambda/feast-subscriptions.zip",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/mobile/feast-subscriptions-PROD-database-lambda",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/PROD/mobile/feast-subscriptions-PROD-database-lambda/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "feastsubscriptionsPRODdatabaselambdaServiceRoleDefaultPolicyCAE476A6",
        "Roles": [
          {
            "Ref": "feastsubscriptionsPRODdatabaselambdaServiceRoleD4C1286B",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "feastsubscriptionsPRODdatabaselambdafeastsubscriptionsPRODdatabaselambdaSnsExistingIncomingEventsTopicE3D3F398": {
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "feastsubscriptionsPRODdatabaselambda7E087BB6",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "Region": {
          "Ref": "AWS::Region",
        },
        "TopicArn": {
          "Fn::Join": [
            "",
            [
              "arn:aws:sns:",
              {
                "Ref": "AWS::Region",
              },
              ":",
              {
                "Ref": "AWS::AccountId",
              },
              ":",
              {
                "Fn::GetAtt": [
                  "feastsubscriptionsPRODsnsqueue58434470",
                  "QueueName",
                ],
              },
            ],
          ],
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "feastsubscriptionsPRODsnsqueue58434470": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "MessageRetentionPeriod": 1209600,
        "QueueName": "feast-subscriptions-PROD-queue",
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/mobile-purchases",
          },
          {
            "Key": "Stack",
            "Value": "mobile",
          },
          {
            "Key": "Stage",
            "Value": "PROD",
          },
        ],
      },
      "Type": "AWS::SQS::Queue",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
